crates/rattler_conda_types/src/match_spec/parse.rs

```diff
@@ -163,9 +165,17 @@ fn parse_bracket_vec_into_components(
     bracket: BracketVec,
     match_spec: MatchSpec,
 ) -> Result<MatchSpec, ParseMatchSpecError> {
-    if let Some(&(key, _)) = bracket.first() {
-        // TODO: not supported yet
-        return Err(ParseMatchSpecError::InvalidBracketKey(key.to_owned()));
+    let mut match_spec = match_spec;
+
+    for elem in bracket {
+        let (key, value) = elem;
+        match key {
+            "version" => match_spec.version = Some(VersionSpec::from_str(value)?),
+            "build" => match_spec.build = Some(value.to_string()),
+            "build_number" => match_spec.build_number = Some(value.parse()?),
+            "fn" => match_spec.file_name = Some(value.to_string()),
+            _ => Err(ParseMatchSpecError::InvalidBracketKey(key.to_owned()))?,
```

Comment:
we should add `sha256` and `md5` as in the docs. I think `url` might also be a field that `conda` supports.